src/prelude/internals.mo

```diff
@@ -581,7 +581,7 @@ func @timer_helper() : async () {
 
 var @lastTimerId = 0;
 
-func @setTimer(delayNanos : Nat64, recurring : Bool, job : () -> async ()) : (id : Nat) {
+func @setTimer<system>(delayNanos : Nat64, recurring : Bool, job : () -> async ()) : (id : Nat) {
```

Comment:
Maybe we could also declare `@cancelTimer` with `system`.

Reply:
Yep!

Reply:
I don’t think `cancelTimer` needs the protection. It cannot transfer anything out, and even an `cancelTimer<system>` would go undetected in library `async` code.

Reply:
@ggreif thinks we need not - won't fight for it.